metadata/src/main/java/edu/unc/lib/dl/util/URIUtil.java

```diff
@@ -52,14 +52,16 @@ public static String join(String base, String... segments) {
 			}
 		}
 
-		for (String segment : segments) {
-			if (segment.charAt(0) != '/' && pathBuilder.length() > 0) {
-				pathBuilder.append('/');
-			}
-			if (segment.charAt(segment.length() - 1) == '/') {
-				pathBuilder.append(segment.substring(0, segment.length() - 1));
-			} else {
-				pathBuilder.append(segment);
+		if (segments[0] != null) {
```

Comment:
This is to catch the case were someone calls UriUtil.join(base, null) I take it?  I wonder if instead we should do something like:
```
for (String segment : segments) {
    if (segment[i] == null) {
        continue;
    }
    ...
```

It would be helpful to cover this in the UriUtil unit test as well